RecoVertex/BeamSpotProducer/python/BeamSpot_cff.py

```diff
@@ -3,9 +3,10 @@
 from RecoVertex.BeamSpotProducer.BeamSpot_cfi import *
 from RecoVertex.BeamSpotProducer.beamSpotToCUDA_cfi import beamSpotToCUDA as _beamSpotToCUDA
 
-offlineBeamSpotCUDA = _beamSpotToCUDA.clone()
+offlineBeamSpotTask = cms.Task(offlineBeamSpot)
 
-offlineBeamSpotTask = cms.Task(
-    offlineBeamSpot,
-    offlineBeamSpotCUDA
-)
+from Configuration.ProcessModifiers.gpu_cff import gpu
+offlineBeamSpotCUDA = _beamSpotToCUDA.clone()
```

Comment:
I'm not suggesting to change this, just trying to understand: would it be equivalent to replace
```python
from RecoVertex.BeamSpotProducer.beamSpotToCUDA_cfi import beamSpotToCUDA as _beamSpotToCUDA
offlineBeamSpotCUDA = _beamSpotToCUDA.clone()
```
with
```python
from RecoVertex.BeamSpotProducer.beamSpotToCUDA_cfi import beamSpotToCUDA as offlineBeamSpotCUDA
```
?

Reply:
The two are not equivalent. The first one creates a copy/clone of `RecoVertex.BeamSpotProducer.beamSpotToCUDA_cfi.beamSpotToCUDA`, so if the `BeamSpot_cff.py` would do something along
```
offlineBeamSpotCUDA.src = "foo"
```
that change does not propagate to other configurations making use of `RecoVertex.BeamSpotProducer.beamSpotToCUDA_cfi.beamSpotToCUDA`.

The second one uses the very same object as `RecoVertex.BeamSpotProducer.beamSpotToCUDA_cfi.beamSpotToCUDA`, and any changes to `offlineBeamSpotCUDA` **do** propagate to other configurations making use of `RecoVertex.BeamSpotProducer.beamSpotToCUDA_cfi.beamSpotToCUDA`, which could be perceived as unexpected.

In this specific case there is little practical difference, so the choice of cloning is more of a following the recommended general pattern (and also protects for the case that someone else would use the second approach and modify a parameter).